src/pages/iv-materials-tfr/Form/MaterialsTransferForm.js

```diff
@@ -0,0 +1,1018 @@
+    } else if (!formik.values.notificationGroupId) {
+      const data = {
+        recordId: formik.values.recordId,
+        functionId: SystemFunction.MaterialTransfer,
+        notificationGroupId: 0
+      }
+
+      await postRequest({
+        extension: AccessControlRepository.Notification.del,
+        record: JSON.stringify(data)
+      })
+    }
+
+    toast.success(platformLabels.Posted)
+    invalidate()
+    await refetchForm(formik.values.recordId)
+  }
+
+  const onUnpost = async () => {
+    const copy = { ...formik.values }
+    delete copy.transfers
+    delete copy.notificationGroupId
+    copy.date = !!copy.date ? formatDateToApi(copy.date) : null
+    copy.closedDate = !!copy.closedDate ? formatDateToApi(copy.closedDate) : null
+    copy.receivedDate = !!copy.receivedDate ? formatDateToApi(copy.receivedDate) : null
+
+    await postRequest({
+      extension: InventoryRepository.MaterialsTransfer.unpost,
+      record: JSON.stringify(copy)
+    })
+
+    toast.success(platformLabels.Unposted)
+    invalidate()
+    await refetchForm(formik.values.recordId)
+  }
+
+  const actions = [
+    {
+      key: 'RecordRemarks',
+      condition: true,
+      onClick: 'onRecordRemarks',
+      disabled: !editMode
+    },
+    {
+      key: 'GL',
+      condition: true,
+      onClick: 'onClickGL',
+      disabled: !editMode
+    },
+    {
+      key: 'IV',
+      condition: true,
+      onClick: 'onInventoryTransaction',
+      disabled: !editMode || !isPosted
+    },
+    {
+      key: 'Locked',
+      condition: isPosted,
+      onClick: 'onUnpostConfirmation',
+      onSuccess: onUnpost,
+      disabled: !editMode || !isClosed || formik.values.isVerified
+    },
+    {
+      key: 'Unlocked',
+      condition: !isPosted,
+      onClick: onPost,
+      disabled: !editMode || !isClosed || formik.values.isVerified
+    },
+    {
+      key: 'Close',
+      condition: !isClosed,
+      onClick: () => onClose(formik.values.recordId),
+      disabled: isClosed || !editMode || isPosted
+    },
+    {
+      key: 'Reopen',
+      condition: isClosed,
+      onClick: onReopen,
+      disabled: !isClosed || !editMode || isPosted
+    },
+    {
+      key: 'WorkFlow',
+      condition: true,
+      onClick: onWorkFlowClick,
+      disabled: !editMode
+    }
+  ]
+
+  async function getNotificationData(recordId) {
+    return await getRequest({
+      extension: AccessControlRepository.Notification.get,
+      parameters: `_recordId=${recordId}&_functionId=${SystemFunction.MaterialTransfer}`
+    })
+  }
+
+  async function getDataGrid(recordId) {
+    return await getRequest({
+      extension: InventoryRepository.MaterialsTransferItems.qry,
+      parameters: `_transferId=${recordId}&_functionId=${SystemFunction.MaterialTransfer}`
+    })
+  }
+
+  useEffect(() => {
+    if (documentType?.dtId) formik.setFieldValue('dtId', documentType.dtId)
+
+    if (documentType?.dtId) getDTD(documentType?.dtId)
+  }, [])
+
+  useEffect(() => {
```

Comment:
merge 2 useEffects into one, since both has no dependencies and will render initially